src/ProgressOnderwijsUtils.Analyzers/AnalyzerUtils.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace ProgressOnderwijsUtils.Analyzers
+{
+    static class AnalyzerUtils
+    {
+        public static bool IsSymbolOfType(this ITypeSymbol? symbol, TypeKind kind, string name, params string[] ns)
+        {
+            if (symbol?.TypeKind != kind) {
+                return false;
+            } else if (symbol.Name != name) {
+                return false;
+            } else {
+                var cns = symbol.ContainingNamespace;
+                // ReSharper disable once ForCanBeConvertedToForeach

```

Comment:
Waarom?

Reply:
Denk dat deze `for` net iets sneller is dan de `foreach`.

Reply:
Ik denk niet dat het uitmaakt.

Reply:
Een for is boeiend sneller dan een foreach mits het om grote value-types gaat en je bij de hot-spot van je code zit.

Linq darentegen is echt vaak veel trager.

Als je hier iets wilt optimaliseren, dan vermoedelijk vermijd het gebruik van params string[], want dat is elke aanroep een array allocation.

Reply:
Alternatief zou een static readonly Matcher  m = ...hier wel arr alloc... zijn en dan de array dus hergebruiken.

Reply:
Nog een voordeel daarvan is dat je de "gewonere" syntax "ProgressOnderwijsUtils.Collections.Maybe" nog vrij gemakkelijk can ondersteunen.

Reply:
> 
> 
> Alternatief zou een static readonly Matcher m = ...hier wel arr alloc... zijn en dan de array dus hergebruiken.

De analyzer draait concurrent, dus dat lijkt me juist niet handig. Dan heb je een TLS nodig of zo iets.

Verder snap ik niet precies wat je bedoelt? Dat we de 'normalere' string-parm gaan splitsen op de punt of zo iets? Dat is toch juist trager?

Reply:
static readonly initializers zijn "vanzelf" thread-safe, ze worden gedraaid voordat een willekeurige thread die code mag gebruiken.  Daarom is het ook een redelijk makkelijke manier om wat preprocessing te doen waar alle aanroepers baat bij hebben.

De data moet natuurljik wel immutable zijn, maar dat is hoe dan ook makkelijker.  Thread-local-storage is eigenlijk typisch alleen van belang voor mutatie.  (en nog wat microoptimalisaties waar het niet nodig is maar sneller kan zijn, bijv. zoals met regexes, maar dan ben je wel al vrij non-obvious bezig).

Reply:
Ik snap niet wat je bedoelt ben ik bang. Wat is die 'Matcher' dan?

Reply:
#442